test/slh_dsa_test.c

```diff
@@ -471,18 +473,26 @@ static int slh_dsa_deterministic_usage_test(void)
             || !TEST_int_eq(EVP_PKEY_sign_message_init(sctx, sig_alg, params), 1))
         goto err;
 
+    if (!TEST_ptr(dupctx = EVP_PKEY_CTX_dup(sctx)))
+        goto err;
+
     /* Determine the size of the signature & allocate space */
     if (!TEST_int_eq(EVP_PKEY_sign(sctx, NULL, &sig_len, msg, msg_len), 1)
             || !TEST_ptr(sig = OPENSSL_malloc(sig_len))
-            || !TEST_int_eq(EVP_PKEY_sign(sctx, sig, &sig_len, msg, msg_len), 1))
+            || !TEST_int_eq(EVP_PKEY_sign(sctx, sig, &sig_len, msg, msg_len), 1)
+            || !TEST_int_eq(EVP_PKEY_sign(dupctx, sig, &sig_len, msg, msg_len), 1))
```

Comment:
Not essential, but perhaps these should use separate signature buffers that can then be separately verified.